docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/Cache.cs

```diff
@@ -0,0 +1,79 @@
+            // This iteration will be longer than subsequent calls, as the dataset is being accessed and stored for later.
+            // Note that this operation may be relatively quick, as the system may have cached the file.
+            (lines, columnAverage, elapsedSeconds) = TimeToScanIDataView(mlContext, cachedData);
+            Console.WriteLine($"Lines={lines}, averageOfColumn0={columnAverage:0.00} and took {elapsedSeconds} seconds.");
+            // Expected output (time is approximate):
+            // Lines=506, averageOfColumn0=564.17 and took 0.056 seconds.
+
+            // Time how long it takes to page through the records now that the data is cached. After the first iteration that caches the IDataView,
+            // future iterations, like this one, are faster because they are pulling from data cached in memory.
+            (lines, columnAverage, elapsedSeconds) = TimeToScanIDataView(mlContext, cachedData);
+            Console.WriteLine($"Lines={lines}, averageOfColumn0={columnAverage:0.00} and took {elapsedSeconds} seconds.");
+            // Expected output (time is approximate):
+            // Lines=506, averageOfColumn0=564.17 and took 0.006 seconds.
+        }
+
+        private static (int lines, double columnAverage, double elapsedSeconds) TimeToScanIDataView(MLContext mlContext, IDataView data)
+        {
+            int lines = 0;
+            double columnAverage = 0.0;
+            var enumerable = mlContext.CreateEnumerable<HousingRegression>(data, reuseRowObject: true);
+            var watch = System.Diagnostics.Stopwatch.StartNew();
+            foreach (var row in enumerable)
+            {
+                lines++;
+                columnAverage += row.MedianHomeValue + row.CrimesPerCapita + row.PercentResidental + row.PercentNonRetail + row.CharlesRiver 
+                    + row.NitricOxides + row.RoomsPerDwelling + row.PercentPre40s + row.EmploymentDistance 
+                    + row.HighwayDistance + row.TaxRate + row.TeacherRatio;
+            }
+            watch.Stop();
+            columnAverage /= lines;
+            var elapsedSeconds = watch.ElapsedMilliseconds / 1000.0;
+
+            return (lines, columnAverage, elapsedSeconds);
+        }
+
+        private class HousingRegression
+        {
+            public float MedianHomeValue { get; set; }
+            public float CrimesPerCapita { get; set; }
+            public float PercentResidental { get; set; }
+            public float PercentNonRetail { get; set; }
+            public float CharlesRiver { get; set; }
+            public float NitricOxides { get; set; }
+            public float RoomsPerDwelling { get; set; }
+            public float PercentPre40s { get; set; }
+            public float EmploymentDistance { get; set; }
+            public float HighwayDistance { get; set; }
+            public float TaxRate { get; set; }
+            public float TeacherRatio { get; set; }
+        }       
```

Comment:
we've been trying to keep the sample support code here:

\Microsoft.ML.SamplesUtils\SamplesDatasetUtils.cs

so the samples display shorter, and more to the point. Maybe we can have this class there too.

Reply:
Done!

Reply:
Done!

---
In reply to: [253741427](https://github.com/dotnet/machinelearning/pull/2406#discussion_r253741427) [](ancestors = 253741427)